catchpoint-mixin/targets.libsonnet

```diff
@@ -0,0 +1,683 @@
+local g = import './g.libsonnet';
+local prometheusQuery = g.query.prometheus;
+local commonlib = import 'common-lib/common/main.libsonnet';
+local utils = commonlib.utils {
+  labelsToPanelLegend(labels): std.join(' - ', ['{{%s}}' % [label] for label in labels]),
+};
+
+{
+  new(this): {
+    local vars = this.grafana.variables,
+    local config = this.config,
+    local testNameLabel = config.testNameLabel,
+    local nodeNameLabel = config.nodeNameLabel,
+    local testAndNodeLabel = config.nodeNameLabel + config.testNameLabel,
+
+    topAvgLoadTimeTestName:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'topk(1, avg by (test_name) (avg_over_time(catchpoint_load_time{%(pureTestNameSelector)s}[$__interval:])))' % vars
```

Comment:
@BominRahmani, @mshahzeb  Do you think there would be any value in making this `topk(k, avg[...])` where `k` is changeable by the user as a selector variable on the dashboards? 

(that'd apply throughout, I guess)

Reply:
I originally had this goal in mind, but @JonathanWamsley and I quickly found out that due to how the data is presented, introducing TopK with k selectors clutters up each of the panels to the point its unreadable. Each of the tests or nodes gets recognized as the top value at a specific point in time and if they are the only value that comes in at that period, then they are automatically recognized as a top value. This leads to basically every test or node being recognized as a top value which defeats the purpose of it. Its kind of hard to explain, you can see if you make the query you were talking about above with topk(k, avg[...]) and insert any number above 2 for k

Reply:
I think I see what you mean yeah, with anything other than topk(1) being equivalent to dropping the topk entirely. 
Thanks for explaining!